api/proto/config/appconfig.proto

```diff
@@ -14,45 +14,64 @@ import "config/netconfig.proto";
 
 message InstanceOpsCmd {
   uint32 counter = 2;
-  string opsTime = 4;
+  string opsTime = 4; // Not currently used
 }
 
+// The complete configuration for an Application Instance
+// When changing key fields such as the drives/volumeRefs or the number
+// of interfaces, the controller is required to issue a purge command i.e.,
+// increase the purge counter. Otherwise there wil be an error (The controller
+// can also issue a purge command to re-construct the content of the first
+// drive/volumeRef without any changes.)
+// Some changes such as ACL changes in the interfaces do not require a restart,
+// but all other changes (such as fixedresources and adapters) require a
+// restart command i.e., an increase to the restart counter. The restart counter
+// can also be increased to cause an application instance restart without
+// any other change to the application instance.
```

Comment:
This is a good summary. Wondering if we need to add this information for each field?

// Needs restart for any changes, including cahnge in order of adapters
NetworkAdapter

// adapters - Needs restart for any changes, including cahnge in order of adapters

Reply:
We can improve that as we go along. I just took an initial crack at is since I wanted to clarify purge.

Reply:
Note that doing that requires verifying what EVE actually does with adapters etc.

Reply:
Which might mean we should have some test cases for it. Hence not just editing comments in my mind.